zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/Publisher.java

```diff
@@ -480,38 +488,55 @@ private static void publishFile(
+        for (Map.Entry<String, String> entry : collection.getDescription().publishTransactionIds.entrySet()) {
             Host host = new Host(entry.getKey());
             String transactionId = entry.getValue();
-            try {
-                endPublish(host, "rollback", transactionId, encryptionPassword);
+
+            try (Http http = new Http()) {
+                Endpoint endpoint = new Endpoint(host, ROLLBACK_ENDPOINT)
+                        .setParameter(TRANSACTION_ID_PARAM, transactionId)
+                        .setParameter(ENCRYPTION_PASSWORD_PARAM, encryptionPassword);
+
+                logInfo("PUBLISH: sending rollback transaction request for collection")
+                        .collectionId(collection)
+                        .collectionName(collection)
+                        .hostToTransactionID(collection.getDescription().publishTransactionIds)
+                        .log();
+
+                Response<Result> response = http.post(endpoint, Result.class);
+                checkResponse(response, transactionId, endpoint, null);
+
             } catch (IOException e) {
-                logError(e, "Error rolling back publish transaction:").log();
+                logError(e, "PUBLISH: error rolling back publish transaction")
+                        .collectionId(collection)
+                        .collectionName(collection)
+                        .transactionID(transactionId)
+                        .log();
             }
         }
     }
 
-    static Result endPublish(Host host, String endpointName, String transactionId, String encryptionPassword) throws IOException {
-        Result result;
-        try (Http http = new Http()) {
-            Endpoint endpoint = new Endpoint(host, endpointName)
-                    .setParameter("transactionId", transactionId)
-                    .setParameter("encryptionPassword", encryptionPassword);
-            Response<Result> response = http.post(endpoint, Result.class);
-            checkResponse(response);
-            result = response.body;
-        }
-        return result;
-    }
-
-    static void checkResponse(Response<Result> response) throws IOException {
-
+    static void checkResponse(Response<Result> response, String TransactionID, Endpoint endpoint, String collectionID) throws
+            IOException {
         if (response.statusLine.getStatusCode() != 200) {
             int code = response.statusLine.getStatusCode();
             String reason = response.statusLine.getReasonPhrase();
             String message = response.body != null ? response.body.message : "";
+
+            URI uri = endpoint.url();
+            logWarn("PUBLISH: request was unsuccessful")
+                    .transactionID(TransactionID)
+                    .collectionId(collectionID)
+                    .addParameter("trainHost", uri.getScheme() + "://" + uri.getHost() + ":" + uri.getPort())
```

Comment:
You can pass `uri` in on its own and it will produce the same result, instead of building the string from parts